idemp-rest/WEB-INF/src/org/bandahealth/idempiere/rest/service/db/VisitDBService.java

```diff
@@ -126,9 +127,22 @@ public static String getLastVisitDate(MBPartner_BH patient) {
 
 	@Override
 	public Visit processEntity(String uuid, String docAction) throws Exception {
-		// We need to do something special for completing a sales order - do it
-		// asynchronously
-		if (StringUtil.isNullOrEmpty(docAction) || !docAction.equalsIgnoreCase(DocAction.ACTION_Complete)) {
+		int clientId = Env.getAD_Client_ID(Env.getCtx());
+		String clientIdsForSynchronousProcessingString =
```

Comment:
I was thinking about this in the morning and, in alignment with something @andrewmoko had said previously, can we make it check UUIDs so that we won't have collisions on our other servers?

Reply:
What do you mean like check?

Reply:
Instead of checking against client IDs, can we make it check against client UUIDs?